### PR TITLE
Remove /scripts/build-consumer and documentation

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

With the release of `/snapit` we can remove this script and dependency.

### WHAT is this pull request doing?

- [x] Removes build-consumer script
- [x] Updates documentation

### CODE REVIEW

File: polaris-react/README.md [hunk 1]
```diff
@@ -98,26 +98,7 @@ To test the changes on a mobile or virtual machine, you will need to open the so
...
-In the example above, the build is copied to `polaris-styleguide/node_modules/@shopify/polaris`. And in this case, a rebuild of `polaris-styleguide` is required after copying the `polaris-react` build, but may not be the case for all consuming projects.
-
-```sh
-# Example
-cd ../polaris-styleguide/
-yarn run build:development
-```
-
-Also, when running `yarn install`, copied builds will be overwritten and will require running `yarn run build-consumer PROJECT_DIRECTORY` again.
+To test your changes in a consuming product you can run the `/snapit` command in your pull request to publish a snapshot NPM package for testing.
```
Comment (alex-page) [inline polaris-react/README.md]:
```suggestion
The `/snapit` GitHub comment command in pull requests will publish a snapshot NPM package for testing.
```

File: polaris-react/README.md [hunk 2]
```diff
@@ -98,26 +98,7 @@ To test the changes on a mobile or virtual machine, you will need to open the so
...
-In the example above, the build is copied to `polaris-styleguide/node_modules/@shopify/polaris`. And in this case, a rebuild of `polaris-styleguide` is required after copying the `polaris-react` build, but may not be the case for all consuming projects.
-
-```sh
-# Example
-cd ../polaris-styleguide/
-yarn run build:development
-```
-
-Also, when running `yarn install`, copied builds will be overwritten and will require running `yarn run build-consumer PROJECT_DIRECTORY` again.
+The `/snapit` GitHub comment command in pull requests will publish a snapshot NPM package for testing.
```
Comment (aaronccasanova) [inline polaris-react/README.md]: **suggestion:** Should we add a link to the [snapshot-release docs](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#snapshot-release)?

**Review by aaronccasanova — APPROVED** [review]
🙌

**Review by tmlayton — APPROVED** [review]

File: polaris-react/README.md (hunk 2 above)
Comment (alex-page) [inline polaris-react/README.md]:
```suggestion
The `/snapit` GitHub comment command in pull requests will publish a snapshot NPM package for testing. Read the [release documentation](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#snapshot-release) for more information.
```

**Pull request merged**